api/src/test/integration/com/javadiscord/jdi/core/api/ApplicationRequestTest.java

```diff
@@ -0,0 +1,47 @@
+package com.javadiscord.jdi.core.api;
+
+import com.javadiscord.jdi.core.Guild;
+import com.javadiscord.jdi.core.api.builders.EditCurrentApplicationBuilder;
+import com.javadiscord.jdi.core.models.application.Application;
+import helpers.LiveDiscordHelper;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+public class ApplicationRequestTest {
+
+    private static Guild guild;
+
+    @BeforeAll
+    public static void setup() throws InterruptedException {
+        guild = new LiveDiscordHelper().getGuild();
+    }
+
+    @Test
+    void testGetApplication() throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(1);
+
+        AsyncResponse<Application> asyncResponse = guild.application().getCurrentApplication();
+        asyncResponse.onSuccess(res -> latch.countDown());
```

Comment:
Can you add some assertions please on the data we're receiving? This captures a few things such as the model being correct, endpoint is correct etc. Under the hood we might of accidently done `/channel` for `getCurrentApplication();` so it's good to ensure that :)

Reply:
well its kinda my question:
> How exactly should I test the Edit Current Application endpoint? These fields that I change are pretty critical.

What exactly should I test? We would first make some default state that we can assume but idk if there is a better solution

Reply:
(and thats kinda why its a draft pr)

Reply:
Just assert the fields are what you expect them